src/ucp/wireup/wireup_cm.c

```diff
@@ -643,7 +647,7 @@ static void ucp_cm_client_connect_cb(uct_ep_h uct_cm_ep, void *arg,
               ucp_ep, ucp_ep->flags, ucp_ep->cfg_index,
               ucs_status_string(status));
 
-    UCP_CM_CB_CHECK_UCP_EP(ucp_ep, uct_cm_ep, return);
+    UCP_EP_CM_CALLBACK_ENTER(ucp_ep, uct_cm_ep, return);
```

Comment:
are we relying on connect_cb being called only from the async thread?

Reply:
yes, we rely on this

Reply:
seems like this is wrong
https://github.com/openucx/ucx/blob/fc1c37f131bcb8c6d27991c228dbb4a5e4b3b206/src/uct/api/uct.h#L559

Reply:
for any async callback

Reply:
discussed offline
will do ASYNC block/unblock in CM callbacks in the separate PR